Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.cpp

```diff
@@ -26,22 +26,21 @@ AZ_POP_DISABLE_WARNING
 
 namespace AzToolsFramework
 {
-    AZ::IO::Path GetAbsolutePathFromRelativePath(const AZ::IO::Path& relativePath, const AZStd::string& productExtension)
+    AZ::IO::Path GetAbsolutePathFromRelativePath(const AZ::IO::Path& relativePath)
     {
-        // The GetFullSourcePathFromRelativeProductPath API only works on product paths,
-        // so we need to append the product extension to try and find it
-        AZStd::string productRelativePath = relativePath.Native() + productExtension;
-
-        AZStd::string fullPath;
+        // First check if this source relative path already exists and is in an asset safe folder
+        AZ::Data::AssetInfo info;
+        AZStd::string watchFolder;
         bool fullPathIsValid = false;
-        AssetSystemRequestBus::BroadcastResult(
-            fullPathIsValid, &AssetSystemRequestBus::Events::GetFullSourcePathFromRelativeProductPath, productRelativePath, fullPath);
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            fullPathIsValid, &AzToolsFramework::AssetSystemRequestBus::Events::GetSourceInfoBySourcePath, relativePath.c_str(), info,
+            watchFolder);
```

Comment:
This function works for absolute and relative paths

Reply:
Correct, in this context though we only have the relative path since that's what we serialize.